components/sidebar/sidebar.tsx

```diff
@@ -49,11 +49,22 @@ export default class Sidebar extends React.PureComponent<Props, State> {
             showMoreChannelsModal: false,
             showNewChannelModal: false,
             showCreateCategoryModal: false,
-            newCategoryIds: [],
             isDragging: false,
         };
     }
 
+    componentDidMount() {
+        if (this.props.teamId) {
+            this.props.actions.fetchMyCategories(this.props.teamId);
+        }
+    }
+
+    componentDidUpdate(prevProps: Props) {
+        if (this.props.teamId && prevProps.teamId !== this.props.teamId) {
+            this.props.actions.fetchMyCategories(this.props.teamId);
```

Comment:
I suppose, but do we have websocket events to bring in categories? I'm thinking of the case where you have 2 sessions open and you create a category on one, and expect it to propagate to the other.

In this case you'd still have to switch teams, but it will always make sure they're updated. It would probably be better to have a websocket event.

Reply:
Yeah, we'll have websocket events to do that when categories are created/edited from other sessions.

There's a chance that it could get out of sync if the websocket goes down, but I don't think the sidebar is critical enough that we have to worry too much about that now.